stdlib/public/core/CString.swift

```diff
@@ -60,6 +60,87 @@ extension String {
       UnsafeBufferPointer(start: cString, count: len)).0
   }
 
+  /// Creates a new String with the specified capacity in UTF-8 code units then
+  /// calls the given closure with a buffer covering the String's uninitialized
+  /// memory.
+  ///
+  /// The closure should return the number of code units that are initialized,
+  /// or nil if it was unable to initialize the buffer (for example if the
+  /// requested capacity ended up being too small for the data).
+  ///
+  /// This initializer does not try to repair ill-formed UTF-8 code unit
+  /// sequences. If any are found, the result of the initializer is `nil`.
```

Comment:
Yeah, probably the sane choice here if it's failable anyways

Reply:
One thing I'm slightly concerned about is that currently tagged -> small bridging doesn't validate OR repair UTF8, and I'm not sure how to maintain that without making the API too sharp edged for normal use…